### PR TITLE
Use a helper script to built builtin.quirk.gz

### PR DESCRIPTION
**Pull request by hughsie**
Stripping out the spaces and comments makes the file 33% smaller.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

File: src/meson.build [hunk 1]
```diff
@@ -102,10 +101,19 @@ fwupdutil = library(
...
 )
 
-if build_daemon
-install_data(['org.freedesktop.fwupd.xml'],
-  install_dir: join_paths(datadir, 'dbus-1', 'interfaces')
+dbus_interface = custom_target('fwupd-generate-dbus-interface',
+  input : 'org.freedesktop.fwupd.xml',
+  output : 'org.freedesktop.fwupd.xml',
+  command : [
+    join_paths(meson.project_source_root(), 'contrib', 'generate-dbus-interface.py'),
```
Comment (superm1) [inline src/meson.build:108]: This is assuming that the shebang in the top works which I thought is a no-no from past issues.  So aren't you missing a python launcher for this?

Comment (superm1) [reply src/meson.build:108]: Also - we shouldn't be putting python scripts that are mandatory for the build process into `contrib`.

Comment (hughsie) [reply src/meson.build:108]: >  So aren't you missing a python launcher for this?

100% yes, good catch.

Comment (hughsie) [reply src/meson.build:108]: > python scripts that are mandatory for the build

I couldn't see a better option, and I think we need to move quite a few around now. libfwupd or libfwupdplugin don't feel right -- maybe `data`?

File: contrib/generate-dbus-interface.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-r",
+        "--releases",
+        type=int,
+        default=5,
+    )
+    parser.add_argument(
+        "filename_src", action="store", type=str, help="metainfo source"
```
Comment (superm1) [inline contrib/generate-dbus-interface.py:30]: just as a personal preference, why not make this `src` and `dst`?  It seems needlessly long.

File: contrib/generate-dbus-interface.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-r",
+        "--releases",
+        type=int,
+        default=5,
```
Comment (superm1) [inline contrib/generate-dbus-interface.py:27]: eh?

File: contrib/generate-dbus-interface.py (hunk 2 above)
Comment (hughsie) [reply contrib/generate-dbus-interface.py:30]: Agreed, done.

File: contrib/generate-dbus-interface.py (hunk 3 above)
Comment (hughsie) [reply contrib/generate-dbus-interface.py:27]: Removed.

File: contrib/generate-dbus-interface.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        root_new = ET.Element('node')
+    #    root_new = tree_new.getroot()
+        root_new.tag = "node"
+        tree_new._setroot(root_new)
+
+        root_new.set('name', "/")
+        n_interface = root.findall("interface")[0]
+        print(n_interface)
+        root_new.append(n_interface)
+    """
```
[the remark this answers is not in the document]
Comment (hughsie) [reply contrib/generate-dbus-interface.py:50]: oopa, removed.

File: contrib/generate-quirk-builtin.py
Comment (superm1) [inline contrib/generate-quirk-builtin.py:1]: move this from contrib

File: src/meson.build (hunk 1 above)
Comment (hughsie) [reply src/meson.build:108]: not `build` -- maybe a `meson-helpers` toplevel directory?

File: meson.build [hunk 5]
```diff
@@ -604,19 +604,15 @@ if build_standalone
...
-  gzip = find_program('gzip')
+  # append all the quirks into one big file and gzip it
   custom_target('builtin-quirk-gz',
-    input: builtin_quirk,
+    input: plugin_quirks,
     output: 'builtin.quirk.gz',
-    capture: true,
-    command: [gzip, '-n', '-k', '--stdout', '@INPUT@'],
+    command: [
+      join_paths(meson.project_source_root(), 'contrib', 'generate-quirk-builtin.py'),
```
[the remark this answers is not in the document]
Comment (hughsie) [reply meson.build:612]: Fixed.

File: src/meson.build (hunk 1 above)
Comment (hughsie) [reply src/meson.build:108]: the problem with `src` is that it's visited _late_ in the build process; if we had something visited before libfwupd we could just do a `mesonhelperdir = dir(''.')` and then refer to the generate python scripts with `file`

Comment (hughsie) [reply src/meson.build]: Okay, fixed.

File: contrib/generate-quirk-builtin.py
Comment (hughsie) [reply contrib/generate-quirk-builtin.py:1]: Done.

**Review by superm1 — APPROVED** [review]

**Pull request merged**